exchanges/huobi/huobi_wrapper.go

```diff
@@ -288,81 +326,189 @@ func (h *HUOBI) Run() {
+		if err != nil {
+			return nil, err
 		}
-		pairs = append(pairs, symbols[x].BaseCurrency+
-			format.Delimiter+
-			symbols[x].QuoteCurrency)
-	}
 
+		for x := range symbols {
+			if symbols[x].State != "online" {
+				continue
+			}
+			pairs = append(pairs, symbols[x].BaseCurrency+
+				format.Delimiter+
+				symbols[x].QuoteCurrency)
+		}
+
+	case asset.CoinMarginedFutures:
+		symbols, err := h.GetSwapMarkets(currency.Pair{})
+		if err != nil {
+			return nil, err
+		}
+
+		for z := range symbols {
+			if symbols[z].ContractStatus == 1 {
+				pairs = append(pairs, symbols[z].ContractCode)
+			}
+		}
+
+	case asset.Futures:
+		symbols, err := h.FGetContractInfo("", "", currency.Pair{})
+		if err != nil {
+			return nil, err
+		}
+
+		for c := range symbols.Data {
+			if symbols.Data[c].ContractStatus == 1 {
+				pairs = append(pairs, symbols.Data[c].ContractCode)
+			}
+		}
+	}
 	return pairs, nil
 }
 
 // UpdateTradablePairs updates the exchanges available pairs and stores
 // them in the exchanges config
 func (h *HUOBI) UpdateTradablePairs(forceUpdate bool) error {
-	pairs, err := h.FetchTradablePairs(asset.Spot)
+	spotPairs, err := h.FetchTradablePairs(asset.Spot)
+	if err != nil {
+		return err
+	}
+	p, err := currency.NewPairsFromStrings(spotPairs)
+	if err != nil {
+		return err
+	}
+	err = h.UpdatePairs(p, asset.Spot, false, forceUpdate)
 	if err != nil {
 		return err
 	}
 
-	p, err := currency.NewPairsFromStrings(pairs)
+	futuresPairs, err := h.FetchTradablePairs(asset.Futures)
+	if err != nil {
+		return err
+	}
+	fp, err := currency.NewPairsFromStrings(futuresPairs)
+	if err != nil {
+		return err
+	}
+	err = h.UpdatePairs(fp, asset.Futures, false, forceUpdate)
 	if err != nil {
 		return err
 	}
-	return h.UpdatePairs(p, asset.Spot, false, forceUpdate)
-}
 
-// UpdateTicker updates and returns the ticker for a currency pair
-func (h *HUOBI) UpdateTicker(p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
-	tickers, err := h.GetTickers()
+	coinmarginedFuturesPairs, err := h.FetchTradablePairs(asset.CoinMarginedFutures)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	pairs, err := h.GetEnabledPairs(assetType)
+	cp, err := currency.NewPairsFromStrings(coinmarginedFuturesPairs)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	for i := range pairs {
-		for j := range tickers.Data {
-			pairFmt, err := h.FormatExchangeCurrency(pairs[i], assetType)
-			if err != nil {
-				return nil, err
-			}
+	return h.UpdatePairs(cp, asset.CoinMarginedFutures, false, forceUpdate)
+}
 
-			if pairFmt.String() != tickers.Data[j].Symbol {
-				continue
+// UpdateTicker updates and returns the ticker for a currency pair
+func (h *HUOBI) UpdateTicker(p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	if !h.SupportsAsset(assetType) {
+		return nil, fmt.Errorf("asset type of %s is not supported by %s", assetType, h.Name)
+	}
+	switch assetType {
+	case asset.Spot:
+		tickers, err := h.GetTickers()
```

Comment:
Carrying on from this comment: https://github.com/thrasher-corp/gocryptotrader/pull/530/files#r538911738
I think what needs to happen is to remove the tickerbatching feature of Huobi here.
So I think what we'll need to do is change this asset.Spot case to use
```
tickers, err := h.GetSwapMarketOverview(p)
		if err != nil {
			return nil, err
		}
		err = ticker.ProcessTicker(&ticker.Price{
			High:         tickers.Tick.High,
			Low:          tickers.Tick.Low,
			Volume:       tickers.Tick.Vol,
			Open:         tickers.Tick.Open,
			Close:        tickers.Tick.Close,
			Pair:         p,
			ExchangeName: h.Name,
			AssetType:    assetType})
		if err != nil {
			return nil, err
		}
```

And then on line 104 of `huobi_wrapper.go`, remove the line `TickerBatching:    true,`

This will ensure that tickers can still be retreived for all currency pairs for all asset types without requiring a refactoring of `syncer.go` or `API.Features` to be asset type specific